File: distros/S/Solstice/examples/webservice_example.php
<?php

$private_key = '12345';
$public_id = 'mcrawfor';

$content = '';

$host = "http://inkey.eplt.washington.edu";
$url = "/tools/rest/webq/v1/";
$method = "PUT";


//Build auth key
$date = time();
if($content)
    $content_sha1 = sha1($content);
$to_sign = "$private_key\n$method\n$url\n$date\n$content_sha1";
$auth_key = "SolAuth $public_id:".sha1($to_sign);

//Build Headers
$headers = array();
array_push($headers, "Date: $date");
if($content)
    array_push($headers, "Content-SHA1: $content_sha1");
array_push($headers, "Authorization: $auth_key");

//HTTP work
$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, $host.$url);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
$response = curl_exec($ch);
curl_close($ch);
echo $response;
?> 

File: distros/S/Solstice/examples/webservice_example.pl


use strict;
use warnings;
use Digest::SHA1 qw(sha1_hex);
use LWP::UserAgent;


my $private_key = '12345';
my $public_id = 'mcrawfor';

my $content = '';

my $host = "http://solstice.washington.edu";
my $url = "/tools/rest/webq/v1/?foo=blah";
my $method = "GET";

#build auth key
my $date = localtime;
my $content_sha1 = $content ? sha1_hex($content) : '';
my $to_sign = "$private_key\n$method\n$url\n$date\n$content_sha1";
my $auth_key = "SolAuth $public_id:". sha1_hex($to_sign);


my $ua = LWP::UserAgent->new;
$ua->agent("Solstice Webservices Client/0.1 ");

my $req = HTTP::Request->new($method => $host.$url);
$req->content($content) if $content;
$req->header('Content-SHA1', $content_sha1) if $content;
$req->header('Date', $date);
$req->header('Authorization', $auth_key);

my $res = $ua->request($req);

if ($res->is_success) {
    print $res->content;
}
else {
    print $res->status_line, "\n";
    print $res->content;
}


